whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotSynchronizer.java

```diff
@@ -8,13 +9,31 @@
 @Service
 public class SnapshotSynchronizer {
 
-    void synchronizeDeltasToSnapshot(final List<DeltaChange> changes) {
-        // TODO: apply changes to the snapshot
-        // if 'operation' is '1' (add) then add to snapshot table
-        // if 'operation' is '2' (del) then remove from snapshot table
+    private final SnapshotObjectRepository snapshotObjectRepository;
+    private final DeltaTransformer deltaTransformer;
 
-        // TODO: throw exception (NrtmProcessException?) if operation can't be done coz it
-        //       indicates we've gone out of sync -- re-initialize the source and publish snapshot 1
+    SnapshotSynchronizer(
+        final SnapshotObjectRepository snapshotObjectRepository,
+        final DeltaTransformer deltaTransformer
+    ) {
+        this.snapshotObjectRepository = snapshotObjectRepository;
+        this.deltaTransformer = deltaTransformer;
+    }
+
+    void synchronizeDeltasToSnapshot(final List<DeltaChange> deltas, final long versionId) {
+        for (final DeltaChange change : deltas) {
+            if (change.getAction() == DeltaChange.Action.ADD_MODIFY) {
+                snapshotObjectRepository.insert(
```

Comment:
If its a modify, don't we need op delee or modify the existing object instead of inserting new? Also it should not need any delta change just rpslObject or serial entry is suffice? Serial Entry has the operation enum

Reply:
whois only has add and delete operations, same for the RFC

Reply:
Yes but if we are synchronizing the existing persisted snapshot, there will already be entry of modified object in snapshot json? In this case will there be two entries in snapshot file for same object?

Reply:
yes, you're right. as i mentioned in mattermost there's a missing piece here.